src/base/emu-i386/simx86/interp.c

```diff
@@ -3385,17 +3407,28 @@ intop3b:		{ int op = ArOpsFR[D_MO(opc)];
 			P0 = PC;
 			CODE_FLUSH();
 		}
-		if (CEmuStat & CeS_MOVSS) {
-			/* following non-compiled (sim or protected mode)
-			   mov ss / pop ss only */
-			if (!(CEmuStat & CeS_INHI)) {
-				// directly following mov ss / pop ss
-				CEmuStat |= CeS_INHI;
-				CEmuStat &= ~CeS_TRAP;
-			} else {
-				// instruction after clear unconditionally
-				// even if it's another mov ss / pop ss
-				CEmuStat &= ~(CeS_INHI|CeS_MOVSS);
+		if (CEmuStat & (CeS_MOVSS|CeS_INSTREMU)) {
```

Comment:
Why this instremu changes the logic here?

Reply:
This is just an optimization, instead of
```
if (a)
  ...
if (b)
  ...
```
use
```
if (a||b)
  if (a)
    ...
  if (b)
    ...
```
so the common case in plain cpusim only has one check instead of two.